From 63229aa1abdb98aa69fda9819ed2f40c8082762b Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Thu, 2 Oct 2014 14:34:55 +0200
Subject: [PATCH] localectl: count locale variables from 0, instead of
 VARIABLE_LANG

That way the we make our code safe regarding reordering of the variables
in the enum.
---
 src/locale/localectl.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/locale/localectl.c b/src/locale/localectl.c
index 5917364d7c..9325d9af6d 100644
--- a/src/locale/localectl.c
+++ b/src/locale/localectl.c
@@ -114,7 +114,7 @@ static void print_overriden_variables(void) {
                 goto finish;
         }
 
-        for (j = VARIABLE_LANG; j < _VARIABLE_LC_MAX; j++)
+        for (j = 0; j < _VARIABLE_LC_MAX; j++)
                 if (variables[j]) {
                         if (print_warning) {
                                 printf("Warning: Settings on Kernel Command Line override system locale settings in /etc/locale.conf\n");
@@ -126,7 +126,7 @@ static void print_overriden_variables(void) {
                         printf("                  %s=%s\n", locale_variable_to_string(j), variables[j]);
                 }
  finish:
-        for (j = VARIABLE_LANG; j < _VARIABLE_LC_MAX; j++)
+        for (j = 0; j < _VARIABLE_LC_MAX; j++)
                 free(variables[j]);
 }
 
